documentation/tutorials/stateScripts.md

```diff
@@ -0,0 +1,78 @@
+Note that repeatable migrations are executed as normal.
+
+## Example
+
+Let’s say we have the following 3 migrations:
+
+```
+V1__create_two_tables.sql
+V2__drop_one_table.sql
+V3__alter_column.sql
+```
+
+If we execute these migrations in order we will have gone through the process of creating two tables, dropping one of them and finally altering a column in the remaining table.
+
+Instead of going through this process for every one of our environments, we might decide that it is easier to create the final table as it would be at the end of applying these 3 migrations in order to simplify the SQL we execute as well as our migration history.
+
+To achieve this, we need only create the following script:
+
+```
+S3__create_table.sql
+```
+
+This should contain the SQL that represents our environment after the original 3 migrations are applied. After adding this script to our existing environment, we will notice no difference as shown in the below output after running `flyway info`:
+
+```
++-----------+---------+-------------------+------+---------------------+---------+----------+
+| Category  | Version | Description       | Type | Installed On        | State   | Undoable |
++-----------+---------+-------------------+------+---------------------+---------+----------+
+| Versioned | 1       | create two tables | SQL  |         ...         | Success | No       |
+| Versioned | 2       | drop one table    | SQL  |         ...         | Success | No       |
+| Versioned | 3       | alter column      | SQL  |         ...         | Success | No       |
++-----------+---------+-------------------+------+---------------------+---------+----------+
+```
+
+However, when we come to apply our migrations in a new environment, `flyway info` will show the following output:
+
+```
++-----------+---------+--------------+------------------+--------------+---------+----------+
+| Category  | Version | Description  | Type             | Installed On | State   | Undoable |
++-----------+---------+--------------+------------------+--------------+---------+----------+
+| Versioned | 3       | create table | SQL_STATE_SCRIPT |              | Pending | No       |
++-----------+---------+--------------+------------------+--------------+---------+----------+
+```
+
+Migrations with a version less than or equal to the latest state script's version are ignored as they are considered to be below the baseline version. Running `flyway migrate` will cause just the `S3` script to be applied, and the history table will show this as a result:
```

Comment:
Again I'm not sure we want to use 'below baseline' in this case.

Reply:
see latest commit for all changes